packages/expo-video/android/src/main/java/expo/modules/video/VideoModule.kt

```diff
@@ -386,4 +382,21 @@ class VideoModule : Module() {
       VideoManager.onAppBackgrounded()
     }
   }
+  private fun replaceImpl(
+    ref: VideoPlayer,
+    source: Either<Uri, VideoSource>?
+  ) {
+    val videoSource = source?.let {
+      if (it.`is`(VideoSource::class)) {
+        it.get(VideoSource::class)
+      } else {
+        VideoSource(it.get(Uri::class))
+      }
+    }
+
+    appContext.mainQueue.launch {
+      ref.uncommittedSource = videoSource
+      ref.prepare()
+    }
```

Comment:
In `replaceAsync`, the promise resolves before committing the source, which may not be accurate. We should manually resolve the promise once the `launch` completes. What do you think?